sync3/handler/connstate_live.go

```diff
@@ -197,6 +198,22 @@ func (s *connStateLive) processLiveUpdate(ctx context.Context, up caches.Update,
 			if delta.JoinCountChanged {
 				thisRoom.JoinedCount = roomUpdate.GlobalRoomMetadata().JoinCount
 			}
+
+			if s.lazyCache.IsLazyLoading(roomUpdate.RoomID()) && roomUpdate.GlobalRoomMetadata().TypingEvent != nil {
```

Comment:
This is being done in the wrong place. If you get a typing notification, it appears in the typing extension, whereas this new code only executes if there is an entry in `rooms` (which means there was some kind of event in said room). You need to:
 - Only do this if the typing extension is enabled.
 - Create an entry in `rooms` if one doesn't exist already, for inserting the room member.